ext/http/lib.rs

```diff
@@ -664,6 +665,57 @@ async fn op_http_write_headers(
   }
 }
 
+#[op]
+async fn op_http_write_resource(
+  state: Rc<RefCell<OpState>>,
+  rid: ResourceId,
+  stream: ResourceId,
+) -> Result<(), AnyError> {
+  let http_stream = state
+    .borrow()
+    .resource_table
+    .get::<HttpStreamResource>(rid)?;
+  let mut wr = RcRef::map(&http_stream, |r| &r.wr).borrow_mut().await;
+  let resource = state.borrow().resource_table.get_any(stream)?;
+  loop {
+    let body_tx = match &mut *wr {
+      HttpResponseWriter::Body(body_tx) => body_tx,
+      HttpResponseWriter::Headers(_) => {
+        return Err(http_error("no response headers"))
+      }
+      HttpResponseWriter::Closed => {
+        return Err(http_error("response already completed"))
+      }
+    };
+
+    let mut vec = vec![0u8; 16 * 1024];
+    let vec_ptr = vec.as_mut_ptr();
+    let buf = ZeroCopyBuf::new_temp(vec);
+    let nread = resource.clone().read(buf).await?;
```

Comment:
Is the vec stack allocated here? Wouldn't that be a problem across the await boundary?

maybe use [into_boxed_slice](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.into_boxed_slice)?

Reply:
> Is the vec stack allocated here? Wouldn't that be a problem across the await boundary?
> 
> maybe use [into_boxed_slice](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.into_boxed_slice)?

Vecs are always heap allocated, it's wrapped in a ZeroCopyBuf to satisfy the resource read trait sig.

As previously mentioned ZeroCopyBuf is conceptually similar to a Cow enum of Owned/Borrowed, Temp is hackish but I'll clean that up